randomArray.js

```diff
@@ -0,0 +1,17 @@
+const _ = require("underscore")
+
+function mixedArray1 (arr) {
+  const newArr = _.shuffle(arr);
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] === newArr[i]){
+      newArr.splice(i, 1);
+      const randomIndex = Math.floor(Math.random() * newArr.length);
```

Comment:
Instead of picking a random element to replace it with, how about just picking an adjacent element?  I think it might simplify the code and still work.